### ISSUE TITLE
[Actions] Opt-out from auto-cancellation

### ISSUE THREAD

**Issue by schlamar** [go-gitea/gitea#32662]
There are a lot of complains about the new auto-cancellation feature (#25716) as this has a lot of unexpected side effects:

- https://github.com/go-gitea/gitea/issues/24769#issuecomment-2211763264
- https://forum.gitea.com/t/job-stopping-on-new-push/8142

The worst side effect: Cancel does not really work as expected on a Windows runner, as child processes are not terminated: https://gitea.com/gitea/act_runner/issues/549

Please provide an option to disable auto-cancellation feature.

**Comment by michaelhofer-slg** [comment]
I agree, this would be very helpful. We have to start the jobs manually again when merging multiple branches within a short time.

**Comment by lyz-code** [comment]
Just to give some more context on why this is important. We're using gitea actions as part of our gitops workflow, so each commit in main triggers a version bump that updates the changelog and does a tag. To be able to do that it needs to push to main these changes, but the workflow is not able to finish because this push cancels the job.
